components/locid/src/helpers.rs

```diff
@@ -30,10 +31,13 @@ impl<T> ShortVec<T> {
     pub fn push(&mut self, item: T) {
         *self = match core::mem::replace(self, Self::ZeroOne(None)) {
             ShortVec::ZeroOne(None) => ShortVec::ZeroOne(Some(item)),
-            ShortVec::ZeroOne(Some(prev_item)) => ShortVec::Multi(vec![prev_item, item]),
-            ShortVec::Multi(mut items) => {
+            ShortVec::ZeroOne(Some(prev_item)) => {
```

Comment:
Actually the only use of `StoreMut` is in `unicode::Value::try_from_bytes`. I think that method should use a vector and then convert that to a `ShortVec` instead of mutating the `ShortVec`. We could then remove `StoreMut`.

Reply:
Removed `lm_push` and `lm_extend` methods from `StoreMut` trait and refactored `unicode::Value::try_from_bytes` to use `Vec` instead of `ShortVec`.

Reply:
I'm fine with making `icu_locid` not require `StoreMut`, but I don't agree with removing `StoreMut` from the `litemap` crate because that severely reduces the funtionality of that crate, which is a utility crate for the Rust ecosytem, and one that we may want to start using again in ICU4X in the future.